components/ws-daemon/pkg/daemon/cache_reclaim.go

```diff
@@ -116,6 +118,11 @@ func readLimit(memCgroupPath string) (uint64, error) {
 	fn := filepath.Join(string(memCgroupPath), "memory.limit_in_bytes")
 	fc, err := os.ReadFile(fn)
 	if err != nil {
+		// TODO(toru): find out why the file does not exists
+		if errors.Is(err, fs.ErrNotExist) {
```

Comment:
Most likely the file doesn't exist anymore because our "dispatch" mechanism takes too long to cancel the context. I.e. the container is already removed before the dispatch mechanism shuts down this mechanism.

Reply:
Should we have todos in the code? If we have them we should at least create an issue and reference them in the comment.

Reply:
The issue has already been created.
https://github.com/gitpod-io/gitpod/issues/8368

Reply:
I saw it :+1: , but was asking more generally if we should have todos in the code.

Reply:
Thanks for your comment 🙏 It seems quite unnatural to ignore this error. So I'd like to write something about the reason.